src/Serialize.h

```diff
@@ -239,7 +239,7 @@ namespace drafter {
         }
 
         template<typename Collection, typename Functor, typename Arg1, typename Arg2>
-        R operator()(const Collection& collection, Functor &wrapper, Arg1 arg1, Arg2 arg2) const {
+        R operator()(const Collection& collection, Functor &wrapper, Arg1 arg1, Arg2& arg2) const {
```

Comment:
you can send context via `std::ref()` it should work. It is not good idea have template which expect some special cases without specialization.